### ISSUE TITLE
Remove limit to 4000 datapoints for local backtest

### ISSUE THREAD

**Issue by MoreDough** [QuantConnect/Lean#466]
There is no need to limit the number of datapoints in the chart data when a local backtest is run. 

Please remove this limit, or at least provide config setting for it when a local backtest is run.

**Comment by jaredbroad** [comment]
Graphic cards / js browsers can only handle so many points and so much RAM before they choke and crash. We've experimented and found roughly 4000 is a good mid-ground which survives on the majority of browsers. There is no sufficient data-store for high resolution charting so everything is done in ram.

In the web platform we're working on allowing zooming; but this won't be available on the desktop as the data isn't available.

This could be made a config parameter with a default setting of 4000. As its hard coded in the series class this would probably mean removing the check in series and putting it into the result handlers who pull out the data.

**Comment by MoreDough** [comment]
Yes, please provide a config param when running desktop backtesting. I am working on my own charting solution for Lean engine using a charting library I am sure can handle much more than 4000 datapoints.

**Comment by Nucs** [comment]
+1
